plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/assemble/LegendAssembler.kt

```diff
@@ -163,6 +163,28 @@ class LegendAssembler(
 
     companion object {
         private const val DEBUG_DRAWING = jetbrains.datalore.plot.FeatureSwitch.LEGEND_DEBUG_DRAWING
+        fun wrap(text: String): String {
```

Comment:
Could you try to rewrite this function using the `runningFold()` function. Looking to the docs there is a good chance to get linear and simple code.